Automattic/woocommerce-payments
# Expand Stripe Link test coverage

Fixes https://github.com/Automattic/woocommerce-payments/issues/5338

#### Changes proposed in this Pull Request
This PR: 
1. Migrates from jQuery to vanilla JS for Stripe Link implementation
2. Wraps the auto-filling logic in `stripe-link/index.js` inside the `autofill` function for easier testability
3. Covers `stripe-link/index.js` with unit tests
4. Adds unit test for the payment gateway initialization logic
5. Documents testing steps in the critical flows documentation










#### Testing instructions
* Confirm that [this page](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) has the **_Stripe Link_** entry
* Read [testing instructions added to the critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#stripe-link), share your feedback, leave a comment if anything needs to be changed/added in the instructions.
* Manually test Stripe Link functionality on the shortcode and Blocks checkout to confirm no regression after migrating from jQuery to vanilla JS and after wrapping the auto-filling logic into `autofill` function. Ideally, use the fresh instructions from the critical flows but also try to think of more edge-cases.
* Confirm if all the possible flows on the front end are covered by `client/checkout/stripe-link/test/index.test.js` and suggest any additions if it's not true





-------------------


**Post merge**
